File: lib-cli/src/protocol/protocol.sizes.rs
use std::mem;

pub const U8_LEN: usize = mem::size_of::<u8>();
pub const U16_LEN: usize = mem::size_of::<u16>();
pub const U32_LEN: usize = mem::size_of::<u32>();
pub const U64_LEN: usize = mem::size_of::<u64>();
pub const I8_LEN: usize = mem::size_of::<i8>();
pub const I16_LEN: usize = mem::size_of::<i16>();
pub const I32_LEN: usize = mem::size_of::<i32>();
pub const I64_LEN: usize = mem::size_of::<i64>();
pub const F32_LEN: usize = mem::size_of::<f32>();
pub const F64_LEN: usize = mem::size_of::<f64>();
pub const BOOL_LEN: usize = mem::size_of::<bool>();

pub enum ESize {
    U8(u8),
    U16(u16),
    U32(u32),
    U64(u64),
}

File: lib-client/src/protocol/protocol.sizes.ts
const CBits = 8;

export const Sizes = {
    u8: 8 / CBits,
    u16: 16 / CBits,
    u32: 32 / CBits,
    u64: 64 / CBits,
    i8: 8 / CBits,
    i16: 16 / CBits,
    i32: 32 / CBits,
    i64: 64 / CBits,
    f32: 32 / CBits,
    f64: 64 / CBits,
    bool: 1,
}

export const Borders = {
    u8: { min: 0, max: 255 },
    u16: { min: 0, max: 65535 },
    u32: { min: 0, max: 4294967295 },
    u64: { min: 0, max: 18446744073709551615 },
    i8: { min: -128, max: 127 },
    i16: { min: -32768, max: 32767 },
    i32: { min: -2147483648, max: 2147483647 },
    i64: { min: -9223372036854775808, max: 9223372036854775807 },
    f32: 32 / CBits,
    f64: 64 / CBits,
    bool: 1,
}